File: stock-analyzer-frontend/src/types/token.ts
// src/types/token.ts
export interface Token { // <<< Make sure 'export interface Token' is exactly like this
    access_token: string;
    token_type: string;
  }
  
  // This is also used by your backend deps.py if you have TokenData for JWT decoding
  // If not used by authService.ts directly yet, it's fine for it to be here.
  export interface TokenData {
    identifier?: string | null; // Stores the 'sub' from JWT (e.g., user's email)
  }

File: backend/app/schemas/token.py
# app/schemas/token.py
from pydantic import BaseModel
from typing import Optional # Or use `str | None` for Python 3.10+

class Token(BaseModel):
    access_token: str
    token_type: str

class TokenData(BaseModel):
    # This will store the identifier from the JWT (e.g., username or email)
    identifier: Optional[str] = None